helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -568,6 +568,71 @@ public void createEphemeral(final String path, final Object data)
     create(path, data, CreateMode.EPHEMERAL);
   }
 
+  /**
+   * Creates a session aware ephemeral node.
+   * If the ephemeral node is created successfully, returns true.
+   * If session is expired, returns false.
+   * If connection is timed out or interrupted, exception is thrown.
+   *
+   * @param path path of the ephemeral node being created
+   * @param data data of the ephemeral node being created
+   * @param sessionId session id of the zookeeper client
+   * @return <code>true</code> if the input session is not expired and the ephemeral node is created
+   *         successfully; <code>false</code> if the input session is expired.
+   * @throws ZkInterruptedException
+   *           if operation is interrupted, or a required reconnection gets interrupted
+   * @throws IllegalArgumentException
+   *           if called from anything except the ZooKeeper event thread
+   * @throws ZkException
+   *           if any ZooKeeper exception occurs
+   * @throws RuntimeException
+   *           if any other exception occurs
+   */
+  public boolean createEphemeral(final String path, final Object data, final long sessionId)
```

Comment:
Lei's comment: 
```
lei-xia 2 minutes ago
Many of the code in this method is duplicated with the one in create() (line 522). Another option is to extend that method to take a sessionID (also wrap the content of that method as a new private method to avoid expose it to be public API)?
```

My reply: I tried to reduce the duplicate code. There are 2 things to consider:
1. Because the input data is an object, we need to serialize it. If so, we need to change the api parameters like:
create(final String path, `byte[] data`, final List<ACL> acl, final CreateMode mode, Callable<String> callable);
And we will need to change create() implementation, whether or not we want to do?
2. Object data needs to be serialized and passed to the final create().
I will show you the change for the above idea and see if we could accept it.